lib/mongo/auth/user.rb

```diff
@@ -49,6 +49,10 @@ class User
       # @return [ Array<String> ] roles The user roles.
       attr_reader :roles
 
+      # Sets the client key for the user.
+      # @return [ String ] The client key for the user.
+      attr_accessor :client_key
```

Comment:
Let's keep this method private so that if we need to change how the client key is cached as some point, we're not bound to a public API.

You can make this method private, then access it with ``user.send(:client_key)`` or you can access it with ``#instance_variable_get`` on the user.